keps/sig-network/0007-pod-ready++.md

```diff
@@ -0,0 +1,96 @@
+---
+kep-number: 1
+title: Pod Ready++
+authors:
+  - "freehan@"
+owning-sig: sig-network
+participating-sigs:
+  - sig-node
+  - sig-cli
+reviewers:
+  - thockin@
+  - dchen1107@
+approvers:
+  - thockin@
+  - dchen1107@
+editor: freehan@
+creation-date: 2018-04-01
+last-updated: 2018-04-01
+status: provisional
+
+---
+
+# Pod Ready++
+
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a KEP and for highlighting any additional information provided beyond the standard KEP template.
+[Tools for generating][] a table of contents from markdown are available.
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+
+
+## Summary
+
+This proposal aims to add extensibility to pod readiness. Besides container readiness, external feedback can be injected into PodStatus and influence pod readiness. Thus, achieving pod “ready++”. 
```

Comment:
I think a better summary is to recognize that what will really happen is to split the concept of "pod ready" into two pieces, for two different sets of consumers.  The current text focuses one one set of consumers, the "workload controllers" that want "pod ready ++".  The other set of consumers is the load balancers, network policy implementors, and so on that contribute to the "pod ready ++" based on "plain old pod ready".  Remember, a load balancer (for example) has to react to a pod before all the load balancers and whatever have reacted to the pod.  Following is a suggested rewording.

This proposal splits the concept of pod readiness into two concepts, for two distinct sets of consumers.  One set of consumers is the workload controllers and other clients that simply want to know when a pod is fully ready for use.  They care about a concept of readiness that we may call "ready++" and is stricter than the concept that is supported today; ready++ also indicates that all the relevant load balancers, traffic filters, and whatever have been adjusted to account for the pod.  The other set of consumers is those very load balancers, traffic filters, and whatever.  They need to continue to react to today's looser concept of readiness, while contributing to the stricter concept.  For example, a load balancer needs to put a new pod into its backend set without waiting for ready++, while also contributing its own bit into ready++.